pilot/pkg/security/authz/model/generator_test.go

```diff
@@ -204,9 +204,11 @@ func TestGenerator(t *testing.T) {
 			key:   "request.headers[x-foo]",
 			value: "foo",
 			want: yamlPrincipal(t, `
-         header:
-          exactMatch: foo
-          name: x-foo`),
+        header:
```

Comment:
The test for the original behavior should remain. The test for the new feature should be a separate test.